src/imitation/data/types.py

```diff
@@ -15,6 +15,8 @@
 
 T = TypeVar("T")
 
+Pathlike = Union[str, bytes, pathlib.Path]
```

Comment:
This is confusing with `os.PathLike`. Some possibilities:
  - `Pathish`
  - `CouldBePath`
  - `PathCastable`

Reply:
To make matters complicated, we probably want to actually allow `pathlib.PathLike` too! `typeshed` uses (for Python >=3.6):
```python
AnyPath = Union[str, bytes, PathLike[str], PathLike[bytes]]
```

Yuck! But perhaps copying that is best?

https://github.com/python/typeshed/blob/c41f3e9bf946c5cb2fa53ed98380dc56977f82e8/stdlib/2and3/_typeshed/__init__.pyi#L52